pkg/controller/migplan/migplan_controller.go

```diff
@@ -434,6 +451,50 @@ func (r *ReconcileMigPlan) AddRunningLabelForPlan(plan *migapi.MigPlan) error {
 	return nil
 }
 
+func (r *ReconcileMigPlan) AddUpdateLabelFailedMigration(plan *migapi.MigPlan) error {
+
+	failedMigrations := 0
+	needsUpdate := false
+	migrations, err := plan.ListMigrations(r)
+	if err != nil {
+		return liberr.Wrap(err)
+	}
+
+	//Check if there are any associated migrations that are failed
+	for _, m := range migrations {
+		if m.Status.HasCondition(migctl.PlanNotReady) {
+			failedMigrations++
+		}
+	}
+
+	switch {
+	//There are no failed migrations associated with the Migplan
+	case failedMigrations == 0:
+		if plan.Labels != nil {
+			if _, found := plan.Labels[MigplanMigrationFailed]; found {
+				delete(plan.Labels, MigplanMigrationFailed)
+				needsUpdate = true
+			}
+		}
+	//Migplan is associated with Failed migrations
+	case failedMigrations > 0:
+		if plan.Labels == nil {
+			plan.Labels = make(map[string]string)
+		}
+		plan.Labels[MigplanMigrationFailed] = "true"
+		needsUpdate = true
+	}
+
+	//Plan update call
+	if needsUpdate {
```

Comment:
inverse the if to remove the indent level, 

```
        if !needs update{
              return nil
        }
     
	liberr.Wrap(r.Update(context.TODO(), plan))
```

Reply:
Done